### PR TITLE
Added error handling for GetENIipLimit in AllocIPAddresses func

### PR DESCRIPTION
**Pull request by Zyqsempai**
Issue #460 

*Description of changes: Added error handling for GetENIipLimit in AllocIPAddresses func*

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by Zyqsempai** [comment]
@tiffanyfay PTAL

File: pkg/awsutils/awsutils.go [hunk 1]
```diff
@@ -870,7 +870,13 @@ func (cache *EC2InstanceMetadataCache) AllocIPAddresses(eniID string, numIPs int
 	var needIPs = numIPs
 
 	ipLimit, err := cache.GetENIipLimit()
-	if err == nil && ipLimit < needIPs {
+	if err != nil {
+		awsUtilsErrInc("UnknownInstanceType", err)
+		// for unknown instance type, will allocate one ip address at a time
+		ipLimit = 1
```
Comment (mogren) [inline pkg/awsutils/awsutils.go]: On error, we should probably just return it, same as for `getMaxEni()`. I added #485 to remove the unused code where the 1 came from. We shouldn't default to 1 IP, that won't work.

Comment (mogren) [reply pkg/awsutils/awsutils.go]: Btw to clarify. We should call `awsUtilsErrInc("UnknownInstanceType", err)`, but after that just `return err`.

Comment (Zyqsempai) [reply pkg/awsutils/awsutils.go]: @mogren Done

**Pull request merged**